src/smc-webapp/frame-editors/frame-tree/save-button.tsx

```diff
@@ -20,51 +17,63 @@ interface Props {
   no_labels?: boolean;
   size?: string;
   onClick?: Function;
+  show_uncommitted_changes?: boolean;
+  set_show_uncommitted_changes?: Function;
 }
 
-export class SaveButton extends Component<Props, {}> {
-  render(): Rendered {
-    const disabled: boolean =
-      !this.props.has_unsaved_changes ||
-      !!this.props.read_only ||
-      !!this.props.is_public;
+const SaveButtonFC: React.FC<Props> = (props: Props) => {
```

Comment:
Please don't name functional components with FC at the end.  They are clearly functional components because of React.FC, and this conflicts with what everybody else does.  I think it's also probably better to just do:
```
const SaveButtonFC: React.FC<Props> = React.memo((props: Props) => { ...
```
so it's crystal clear right when you're reading the code that memo is being used.